src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -866,6 +867,51 @@ class spell_warr_second_wind_trigger : public SpellScriptLoader
         }
 };
 
+// Shield Wall - 871
+class spell_warr_shield_wall : public SpellScriptLoader
+{
+public:
+    spell_warr_shield_wall() : SpellScriptLoader("spell_warr_shield_wall") { }
+
+    class spell_warr_shield_wall_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_warr_shield_wall_AuraScript);
+
+        void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            if (!GetCaster())
+                return;
+
+            if (Player* player = GetCaster()->ToPlayer())
+            {
+                player->AddAura(SPELL_WARRIOR_SHIELD_OF_WALL, player);   
+            }
+        }
+
+        void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            if (!GetCaster())
+                return;
+
+            if (Player* player = GetCaster()->ToPlayer())
+            {
+                player->RemoveAura(SPELL_WARRIOR_SHIELD_OF_WALL);
```

Comment:
RemoveAurasDueToSpell(SPELL_WARRIOR_SHIELD_OF_WALL);
And i cant see any reason to cast GetCaster() to player